content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+-------------------
+
+Have a look at `Atom <https://atom.io/>`. By default it will run
+``elm-format`` each time you save your file which is really handy.
+
+Take a few moments to make sure your text editor is configured to work
+friendly with Elm files.
+
+
+Starting your first project
+===========================
+
+No need for a boilerplate here, you can just start by running
+``elm-package install`` and approve the plan:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-package install
+
+    Some new packages are needed. Here is the upgrade plan.
+
+      Install:
+        elm-lang/core 5.1.1
+        elm-lang/html 2.0.0
+        elm-lang/virtual-dom 2.0.4
+
+    Do you approve of this plan? [Y/n] Y
+    Starting downloads...
+
+      ● elm-lang/html 2.0.0
+      ● elm-lang/virtual-dom 2.0.4
+      ● elm-lang/core 5.1.1
+
+    Packages configured successfully!
+
+You now have a ``elm-package.json`` in your project as well as a
+``elm-stuff/`` directory that contains libraries ``elm-package``
+installed.
+
+.. code-block:: bash
+
+    ~/tutorial$ tree -L2
+	.
+    ├── elm-package.json
+    └── elm-stuff
+        ├── exact-dependencies.json
+        └── packages
+
+    2 directories, 2 files
+
+
+Creating your first file
+========================
+
+No need for a boilerplate here, to get started you can simply create a
+new file named ``Main.elm`` with the following:
+
+.. code-block:: elm
+
+    import Html
+    main = Html.text "Hello world"
+
+Playing with elm-format
+-----------------------
+
+If your editor is well configured with ``elm-format``, you should see
+this as soon as you save it:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html
+
+
+    main =
+        Html.text "Hello world!"
+
+If not, you can run ``elm-format`` manually on your file:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-format Main.elm
+    This will overwrite the following files to use Elm's preferred style:
+
+        Main.elm
+
+    This cannot be undone! Make sure to back up these files before proceeding.
+
+    Are you sure you want to overwrite these files with formatted versions? (y/n)
+
+    y
+    Processing file Main.elm
+
+To avoid having to answer yes each time you can use the ``--yes`` parameter::
+
+    elm-format --yes Main.elm
+
+
+Opening your app in the browser
+-------------------------------
+
+One way to run your app is to use ``elm-reactor``.
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-reactor
+	elm-reactor 0.18.0
+    Listening on http://localhost:8000
+
+Then open: http://localhost:8000/Main.elm
+
+You are now viewing your Elm app live!!!
+
+
+Learning about the Elm virtual DOM
+==================================
+
+The Elm Virtual DOM, functions to generate HTML are in the ``Html``
+module.
+
+If you want you can import all of it in your module using:
+
+.. code-block:: elm
+
+    import Html exposing (..)
+
+You can then use ``text`` directly for instance:
+
+.. code-block:: elm
+
+    main = text "Hello world"
```

Comment:
Maybe add a note about `(..)` which is not the strict equivalent of `import *`. Elm compiler will prevent any overriden imports, which is a key and makes the feature really enjoyable.